app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt

```diff
@@ -0,0 +1,395 @@
+        withIndex(
+          withId(R.id.slide_title_text_view),
+          2
+        )
+      ).check(matches(withText(getResources().getString(R.string.slide_2_title))))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkSlide2Description_isCorrect() {
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(
+        withIndex(
+          withId(R.id.slide_description_text_view),
+          2
+        )
+      ).check(matches(withText(getResources().getString(R.string.slide_2_description))))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkSlide2Image_isCorrect() {
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(
+        withIndex(
+          withId(R.id.slide_image_view),
+          2
+        )
+      ).check(matches(withTagValue(equalTo(R.drawable.ic_onboarding_2))))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkSlide2_skipButtonIsVisible() {
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(withId(R.id.skip_text_view)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkSlide2_clickSkipButton_shiftsToLastSlide() {
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(withId(R.id.skip_text_view)).perform(click())
+      onView(
+        withIndex(
+          withId(R.id.slide_title_text_view),
+          1
+        )
+      ).check(matches(withText(getResources().getString(R.string.slide_3_title))))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkSlide2_getStartedButtonIsNotVisible() {
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(withId(R.id.get_started_button)).check(matches(not(isDisplayed())))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkSlide3Title_isCorrect() {
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+      onView(
+        withIndex(
+          withId(R.id.slide_title_text_view),
+          2
+        )
+      ).check(matches(withText(getResources().getString(R.string.slide_3_title))))
```

Comment:
You may directly use withText(final int resourceId) instead of getting string from resource.

Reply:
Done. Thanks.